### PR TITLE
fix: #1453

### LINKED ISSUE #1453: Invalid URLs for OpenApi + SwaggerView with relativ url root
If you'd like to report a bug in Flask-Appbuilder, fill out the template below. Provide
any extra information that may be useful

### Environment

Flask-Appbuilder version:
>3

pip freeze output:
unimportant

### Describe the expected results
When using openapi and swagger view with a relativ url root (e.g. localhost/foo/bar/api/v1) urls are not correct.
A possible fix could be the following:

in api/manager.py 
```
 servers=[{"url": url_for("OpenApi.get", version=version)[:-9]}],
 openapi_uri=url_for(OpenApi.__name__ + "." + OpenApi.get.__name__, version=version),
```

Tell us what should happen.

```python
Paste a minimal example that causes the problem.
```

### Describe the actual results

Tell us what happens instead.

```pytb
Paste the full traceback if there was an exception.
```

### PR DESCRIPTION
**Pull request by lnnwvr**
Fixes #1453

### CODE REVIEW

**Review by dpgaspar — CHANGES_REQUESTED** [review]
Thank you. Looks good, but you need to run `black` and `flake8`

**Comment by lnnwvr** [comment]
finally...
fyi: i had some trouble with flake8 and also psycopg2-binary(which should be upgraged) and python 3.8

File: flask_appbuilder/api/manager.py [hunk 1]
```diff
@@ -58,22 +58,23 @@ def _create_api_spec(version):
             openapi_version="3.0.2",
             info=dict(description=current_app.appbuilder.app_name),
             plugins=[MarshmallowPlugin()],
-            servers=[{"url": "/api/{}".format(version)}],
+            servers=[{"url": url_for("OpenApi.get", version=version)[:-9]}],
```
Comment (dpgaspar) [inline flask_appbuilder/api/manager.py:61]: Sorry, what this for `[:-9]` ?

**Review by dpgaspar — CHANGES_REQUESTED** [review]

File: flask_appbuilder/api/manager.py (hunk 1 above)
Comment (lnnwvr) [reply flask_appbuilder/api/manager.py:61]: url_for("OpenApi.get", version=version) returns url_root + /api/v1/_openapi but we just need url_root + /api/v1

Comment (mjpieters) [reply flask_appbuilder/api/manager.py:61]: I’d use `.rpartition("/")[0]`, which splits the string at the last slash and returns everything before that point.

Isn’t there a better source for the API version root URL however?

Comment (dpgaspar) [reply flask_appbuilder/api/manager.py:61]: The class `OpenApi` itself holds the base and the version is passed to this method. The API version can vary on each `BaseApi` class so the source is at that level

Comment (lnnwvr) [reply flask_appbuilder/api/manager.py:61]: so what do we change?

Comment (mjpieters) [reply flask_appbuilder/api/manager.py:61]: Make it

```python
servers=[{"url": url_for("OpenApi.get", version=version).rpartition("/")[0]}],
```

although I'd pull out the expression into a separate variable:

```python
    @staticmethod
    def _create_api_spec(version):
        # The spec produced here applies to the base of this view:
        api_url = url_for("OpenApi.get", version=version).rpartition("/")[0]
        return APISpec(
            title=current_app.appbuilder.app_name,
            version=version,
            openapi_version="3.0.2",
            info=dict(description=current_app.appbuilder.app_name),
            plugins=[MarshmallowPlugin()],
            servers=[{"url": api_url}],
        )
```

Comment (lnnwvr) [reply flask_appbuilder/api/manager.py]: i made the changes, how do i merge the change it into the pull request .. sometimes i dont understand github

**Comment by lnnwvr** [comment]
Its still a issue isnt it @dpgaspar ?

**Pull request closed without merging**